In [ ]:
#import library
import pandas as pd
pd.set_option('display.max_columns', None)
import numpy as np
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

In [ ]:
df = pd.read_csv('/content/marketing_campaign.csv', sep='\t')

In [ ]:
df.head(3)

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   

  Dt_Customer  Recency  MntWines  MntFruits  MntMeatProducts  MntFishProducts  \
0  04-09-2012       58       635         88              546              172   
1  08-03-2014       38        11          1                6                2   
2  21-08-2013       26       426         49              127              111   

   MntSweetProducts  MntGoldProds  NumDealsPurchases  NumWebPurchases  \
0                88            88                  3                8   
1                 1             6                  2                1   
2                21            42                  1                8   

   NumCatalogPurchases  NumStorePurchases  NumWebVisitsMonth  AcceptedCmp3  \
0                   10                  4                  7             0   
1                    1                  2                  5             0   
2                    2                 10                  4             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

# Cleaning Data & Handling Outliers

In [ ]:
# Menghapus baris yang memiliki nilai Income hilang
df = df.dropna(subset=['Income'])
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2216 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2216 non-null   int64  
 1   Year_Birth           2216 non-null   int64  
 2   Education            2216 non-null   object 
 3   Marital_Status       2216 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2216 non-null   int64  
 6   Teenhome             2216 non-null   int64  
 7   Dt_Customer          2216 non-null   object 
 8   Recency              2216 non-null   int64  
 9   MntWines             2216 non-null   int64  
 10  MntFruits            2216 non-null   int64  
 11  MntMeatProducts      2216 non-null   int64  
 12  MntFishProducts      2216 non-null   int64  
 13  MntSweetProducts     2216 non-null   int64  
 14  MntGoldProds         2216 non-null   int64  
 15  NumDealsPurchases    2216 non-null   int64 

In [ ]:
# Menangani outliers di kolom 'Income' menggunakan metode IQR
Q1 = df['Income'].quantile(0.25)
Q3 = df['Income'].quantile(0.75)
IQR = Q3 - Q1

# Menghapus outliers di luar 1.5 * IQR
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

# Mempertahankan data yang berada dalam batas wajar
df = df[(df['Income'] >= lower_bound) & (df['Income'] <= upper_bound)]

In [ ]:
df['Income'].describe()

count      2208.000000
mean      51633.638134
std       20713.376530
min        1730.000000
25%       35196.000000
50%       51301.000000
75%       68289.750000
max      113734.000000
Name: Income, dtype: float64

# No 1

In [ ]:
# Membuat Function untuk Menghitung PMF dari NumWebVisitsMonth
def pmf(column):
    total_observations = len(column)
    value_counts = column.value_counts().sort_index()
    probabilities = value_counts / total_observations
    return probabilities

In [ ]:
# Menghitung PMF untuk NumWebVisitsMonth
pmf_web_visits = pmf(df['NumWebVisitsMonth'])
pmf_web_visits

NumWebVisitsMonth
0     0.002717
1     0.066576
2     0.091033
3     0.091938
4     0.098279
5     0.126359
6     0.151268
7     0.175272
8     0.153986
9     0.037138
10    0.001359
13    0.000453
14    0.000906
17    0.000453
19    0.000906
20    0.001359
Name: count, dtype: float64

- Dari data di atas, jumlah kunjungan web bulanan yang paling sering terjadi adalah antara 5 hingga 8 kunjungan per bulan, dengan 7 kunjungan bulanan memiliki probabilitas tertinggi, yaitu 17.53%.

In [ ]:
# Probabilitas memiliki 5 kunjungan
prob_5_visits = pmf_web_visits.get(5, 0)
prob_more_than_5_visits = pmf_web_visits[pmf_web_visits.index > 5].sum()

In [ ]:
print("Probabilitas 5 kunjungan web bulanan:", prob_5_visits)
print("Probabilitas lebih dari 5 kunjungan web bulanan:", prob_more_than_5_visits)

Probabilitas 5 kunjungan web bulanan: 0.12635869565217392
Probabilitas lebih dari 5 kunjungan web bulanan: 0.5230978260869565


- Dari hasil ini, lebih dari setengah pelanggan memiliki lebih dari 5 kunjungan web bulanan, yang menunjukkan bahwa kebanyakan pelanggan cukup aktif di situs web.

# NO 2

In [ ]:
# Mengelompokkan data berdasarkan pendidikan
groups = [df['Income'][df['Education'] == education] for education in df['Education'].unique()]

In [ ]:
# Melakukan ANOVA
f_stat, p_value = stats.f_oneway(*groups)

In [ ]:
print('F-statistic:', f_stat)
print('P-value:', p_value)

F-statistic: 39.278883537184974
P-value: 8.234399637206527e-32


In [ ]:
if p_value < 0.05:
    print("Hasil: Terdapat perbedaan signifikan antara rata-rata income berdasarkan education (H0 Ditolak)")
else:
    print("Hasil: Tidak ada perbedaan signifikan antara rata-rata income berdasarkan education (H0 Diterima)")

Hasil: Terdapat perbedaan signifikan antara rata-rata income berdasarkan education (H0 Ditolak)


- F-statistic yang tinggi (39.28) mengindikasikan adanya variabilitas yang cukup besar dalam pendapatan berdasarkan tingkat pendidikan.
- P-value yang sangat kecil (lebih kecil dari 0.05, bahkan mendekati nol) menunjukkan bahwa hasil uji ini sangat signifikan. Ini berarti kita dapat menolak hipotesis nol(H0) bahwa pendidikan tidak mempengaruhi pendapatan.  Dengan kata lain, tingkat pendidikan berpengaruh secara signifikan terhadap pendapatan seseorang.
- Pendidikan berpengaruh secara signifikan terhadap pendapatan. Artinya, tingkat pendidikan yang lebih tinggi kemungkinan besar berkorelasi dengan pendapatan yang lebih tinggi.
